### PR TITLE
Shield Foundation Part 3: Bringing it all together

### PR DESCRIPTION
**Pull request by chaosvolt**
Still requires https://github.com/CleverRaven/Cataclysm-DDA/pull/14959 before it'll work fully as intended, but able to prepare it separately.
- Shields given the required combination of flags.
- Head coverage removed from all shields. This was leftover from when shields could only be used as passive protection.
- No longer covers the torso or legs either. It made encumbrance balancing a massive pain in the ass, and allowing ranged attacks to be blocked can be done another day if need be.
- Block techniques changed. Basic shields only get block (WBLOCK_1), bucklers get parry (WBLOCK_2), and larger shields get the shield (WBLOCK_3) technique.
- Coverage percentages altered.
- Encumbrance values altered somewhat, due to shields no longer covering the torso. Generally ramps up consistent with coverage.
  
  | Name | Block Level | Encumbrance | Coverage | Bash/Cut |
  | --- | --- | --- | --- | --- |
  | Wooden Shield | 1 | 12 | 40% | 9/6 |
  | Large Wooden Shield | 3 | 28 | 70% | 9/6 |
  | Heater Shield | 1 | 12 | 40% | 14/9 |
  | Kite Shield | 3 | 24 | 60% | 14/9 |
  | Round Shield | 1 | 16 | 45% | 16/14 |
  | Hoplon | 1 | 20 | 50% | 21/21 |
  | Scutum | 3 | 28 | 70% | 21/18 |
  | Buckler | 2 | 8 | 30% | 27/27 |

### CODE REVIEW

File: data/mods/Medieval_Stuff/shields.json [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+        "id" : "shield_wooden",
+        "name" : "wooden shield",
+        "description" : "A crude wooden shield, lacking any metal or leather renforcement.  Lightweight but not very tough.",
+        "weight" : 2267,
+        "color" : "brown",
+        "covers" : ["ARM_EITHER", "HAND_EITHER"],
+        "price" : 80000,
+        "material" : "wood",
+        "volume" : 15,
+        "cutting" : 0,
```
Comment (Coolthulhu) [inline data/mods/Medieval_Stuff/shields.json]: `"cutting" : 0` is the default value, it doesn't need to be specified.

**Comment by Coolthulhu** [comment]
Not merging now because the hand restriction isn't yet ready.

Some other things I noticed here:
* All shields are STURDY. IRL shields often broke quite a lot
* Adding leather material to shields means they can be sewn to repair them
* Huge coverage on shields is not really OK. 95 on scutum means 95% of hits to the torso - from behind or otherwise - will hit the shield instead.

**Comment by chaosvolt** [comment]
Hmm. Can reduce coverage in that case. Not sure how best to alter the materials for leather-faced shields.

Also, another thought I had. As mentioned, shields covering the torso conflict with backpacks. Should that oddity be retained, should they be given the "WAIST" layer instead, or should they have their torso coverage removed?

They kinda end up being more same-y if they lose their differing body part coverages, and they lose their passive protection that'd be a distinctive feature of the larger shields.

**Comment by Rivet-the-Zombie** [comment]
If the player actively blocks with a shield, then why the kludge that makes them cover anything other than the arm/hand they're worn on?

**Comment by chaosvolt** [comment]
The primary reason is to allow larger shields to provide protection against ranged attacks. Against anything with a gun, it ideally shouldn't do anything that existing medieval armor can't already do.

Since you can't actively block ranged attacks last I checked. That's also why they no longer longer the head. It's now assumed that the coverage is for passive protection, and further assuming that if a player can't actively block ranged attacks, they can't cower behind it, and therefor that passive defense only protects the body parts that would logically be shielded by it.

**Comment by Rivet-the-Zombie** [comment]
I don't think a medieval shield would be much in the way of protection against a bullet in the first place.

**Comment by chaosvolt** [comment]
It probably shouldn't.

Though I would assume it still gets used sensible against corrosive zombies, NPCs with bows, and other limited circumstances. Might not be worth retaining the torso coverage.

**Comment by Coolthulhu** [comment]
Could be reimplemented as a proper active blocking mechanic later on.
See `Creature::deal_projectile_attack` - it already has (partial) dodging.

**Comment by chaosvolt** [comment]
So instead of extending a player ability to NPCs or monsters, we'd be extending a monster ability to the player? o3o

As it is, it's still possible to give them distinctive properties even if I drop the idea of torso coverage, and it does make encumbrance less of a headache to balance...

**Comment by Coolthulhu** [comment]
No, `Creature` is both monsters and players. `deal_projectile_attack` is what is invoked to deal damage on a critter hit by a projectile. It also handles dodging from the projectile, selecting body parts to hit and bonuses from headshots.

You could override the player version of it and add possible blocking there.

**Comment by chaosvolt** [comment]
Ah, I see. Hmm. That might be a more complex addition though.

**Comment by chaosvolt** [comment]
Meep? Thank you for the merge, though there's still https://github.com/CleverRaven/Cataclysm-DDA/pull/14959

As it stands now, players can still load of on two shields. X3

**Comment by Rivet-the-Zombie** [comment]
It hadn't occurred to me to try doing that!

I guess #14959 is next.

**Comment by chaosvolt** [comment]
Doh. Sorry about that. Oh well, there'll be a glorious build where players can wear 2 of every shield while wielding a zweihander. X3

**Comment by Rivet-the-Zombie** [comment]
When do we get shields in the mainline?

**Comment by chaosvolt** [comment]
Huehuehue. Wasn't sure about that. Though I was sorely tempted to add riot shields, those wouldn't be sensible unless blocking ranged attacks was a thing.

**Comment by Rivet-the-Zombie** [comment]
>I was sorely tempted to add riot shields

>unless blocking ranged attacks was a thing

You don't have to address both in the same PR.

**Comment by chaosvolt** [comment]
Hmm, maybe. Would still be useful against a riot of zombies. X3

**Pull request merged**